src/modules/places/places.view.js

```diff
@@ -21,8 +21,9 @@ class PlacesView extends Component {
   @observable officeList = [];
 
   async componentDidMount() {
-    this.officeList = await this.props.officesStore.getAllOffices();
-    this.placeList = await this.props.placesStore.getAllPlaces();
+    const { officesStore, placesStore } = this.props;
+    this.officeList = await officesStore.getAllOffices();
```

Comment:
Right now, you are waiting for one promise, and then for another.
These two can be fetched simultaneously. Try `Promise.all()`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all